src/_shared/js/capi-images.js

```diff
@@ -0,0 +1,80 @@
+// Creates a set of source elements for an image.
+function buildSources (sourceData) {
+
+	let sourcesFragment = document.createDocumentFragment();
+
+	return sourceData.reduce((sources, source) => {
+
+		let hidpi = document.createElement('source');
+		hidpi.media = `(min-width: ${source.minWidth}px) and
+			(-webkit-min-device-pixel-ratio: 1.25),
+			(min-width: ${source.minWidth}px) and (min-resolution: 120dpi)`;
+		hidpi.sizes = source.sizes;
+		hidpi.srcset = `${source.hidpiSrcset}`;
+
+		let lodpi = document.createElement('source');
+		lodpi.media = `(min-width: ${source.minWidth}px)`;
+		lodpi.sizes = source.sizes;
+		lodpi.srcset = `${source.lodpiSrcset}`;
+
+		sources.appendChild(hidpi);
+		sources.appendChild(lodpi);
+		return sources;
+
+	}, sourcesFragment);
+
+}
+
+// Creates a picture element with responsive sources, with fallback for IE.
+function createPicture (imageInfo, imageElem) {
+
+	// Supports responsive images.
+	if ('srcset' in imageElem) {
+
+		let picture = document.createElement('picture');
```

Comment:
Just thought about it, but in [this PR](https://github.com/guardian/commercial-templates/pull/60) I have created a HOF that abstracts the creation of DOM elements. Might be useful.